CameraActivity.java

```diff
@@ -0,0 +1,336 @@
+        cameraKitView.onResume();
+    }
+
+    @Override
+    protected void onPause() {
+        cameraKitView.onPause();
+        super.onPause();
+    }
+
+    @Override
+    protected void onStop() {
+        cameraKitView.onStop();
+        super.onStop();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        cameraKitView.onRequestPermissionsResult(requestCode, permissions, grantResults);
+    }
+
+
+    /**
+     * Stores the captured image into a temporary file useful to pass large data between activities
+     * and returns the file's path.
+     * @param context The reference of the current activity
+     * @param bitmap The captured image to store into the file. Not null or empty.
+     * @param name The name of the file. Not null or empty.
+     * @return The files path
+     * @author Leonardo Rossi
+     */
+    private String tempFileImage(Context context, Bitmap bitmap, String name)
+    {
+
+        File outputDir = context.getCacheDir();
+        File imageFile = new File(outputDir, name + ".jpg");
+
+        OutputStream os;
+        try {
+            os = new FileOutputStream(imageFile);
+            bitmap.compress(Bitmap.CompressFormat.JPEG, 100, os);
+            os.flush();
+            os.close();
+        } catch (Exception e) {
+            Log.e(context.getClass().getSimpleName(), "Error writing file", e);
+        }
+
+        return imageFile.getAbsolutePath();
+    }
+
+
+    /**
+     * Rotate the bitmap image of the angle
+     * @param source the image
+     * @param angle angle of rotation
+     * @return Bitmap image rotated
+     */
+    public static Bitmap rotateImage(Bitmap source, int angle) {
+        Matrix matrix = new Matrix();
+        matrix.postRotate(angle);
+        return Bitmap.createBitmap(source, 0, 0, source.getWidth(), source.getHeight(),
+                matrix, true);
+    }
+
+    /**
+     * Detect the blurriness by appling a Laplacian matrix and evaluating the variance (low variance means an usually blurry image) uses OpenCV
+     * @param bitmap1 The captured image
+     * @return Double value of the variance
+     * @author Leonardo Pratesi
+     */
+    public static double blurValue(Bitmap bitmap1)
+    {
+        double blur;
+        double threshold= 10;
+        Bitmap image = bitmap1;
+        Mat matImage = new Mat();
+        org.opencv.android.Utils.bitmapToMat(image, matImage);
+        Mat destination = new Mat();
+        Mat matGray=new Mat();
+
+        Imgproc.cvtColor(matImage, matGray, Imgproc.COLOR_BGR2GRAY);    //converte immagine in scala di grigi
+        Imgproc.Laplacian(matGray, destination, 3);              //applica prodotto di convoluzione a matGray e lo mette in destination
+        MatOfDouble median = new MatOfDouble();
+        MatOfDouble std= new MatOfDouble();

```

Comment:
Could I suggest a clearer name like "stdDev"? Or "standardDeviation"?

Reply:
Yes i will keep it in mind